### DISCUSSION TITLE
Support for storing Milvus username & password in external secret management tool instead of Etcd.

### DISCUSSION THREAD

**Discussion by lashsingh** [milvus-io/milvus#27106] [Q&A and General discussion]
Currently Milvus stores encryted username & password in Etcd for database authentication. 

The details are mentioned here: https://milvus.io/blog/data-security.md/#How-does-the-Milvus-vector-database-ensure-data-security

I would like to store the user authentication secrets in external secret management tool like Hashicorpt Vault instead of Etcd. 

I would like to know who this can be achieved, any documentation around this will be very helpful.

Thanks for your help in advance.

**Comment by yhmo** [comment]
It is a new requirement that requires vault interfaces to interact with vault and extra configurations to access vault. I don't think we have the resources to implement it.

@xiaofan-luan

**Comment by xiaofan-luan** [comment]
> Currently Milvus stores encryted username & password in Etcd for database authentication.
> 
> The details are mentioned here: https://milvus.io/blog/data-security.md/#How-does-the-Milvus-vector-database-ensure-data-security
> 
> I would like to store the user authentication secrets in external secret management tool like Hashicorpt Vault instead of Etcd.
> 
> I would like to know who this can be achieved, any documentation around this will be very helpful.
> 
> Thanks for your help in advance.

Milvus is opensource project and don't want to introduce any other outside dependencies.
Sorry about that.
ETCD has encrption and should be good enough for most cases.

For Zilliz cloud we use AWS KMS to store the key/password

**Reply by lashsingh** [reply]
Ok, understood. Thanks.
